### PR TITLE
[MERGE] web,*: editable list view: column widths

### PR DESCRIPTION
**Pull request by aab-odoo**
This PR improves the display of editable of list views.

Instead of always hardcoding column widths depending of the field types, we let
the browser compute them as much as possible, i.e. as soon as there are records,
and we then freeze the widths s.t. it doesn't flicker when switching a row in edition.
However, we keep the fixed widths heuristic when there is no record to display, as
in this case the browser doesn't have enough information to compute the widths.

In the latter case, we also tweaked the heuristic s.t. all field types having a relative
width now have the same width factor (i.e. the available space is uniformly shared
between them).

Task 2011587

### CODE REVIEW

**Comment by aab-odoo** [comment]
robodoo rebase-merge

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 1]
```diff
@@ -491,6 +492,13 @@ ListRenderer.include({
      * @override
      */
     updateState: function (state, params) {
+        this.oldColumns = this.columns;
+        this.columnWdiths = false;
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: hmm, read carefully this line... Maybe it is useless, maybe a test is missing...

File: addons/web/static/tests/fields/relational_fields/field_one2many_tests.js [hunk 2]
```diff
@@ -8541,6 +8541,70 @@ QUnit.module('fields', {}, function () {
...
+                                '</tree>' +
+                            '</field>' +
+                        '</form>',
+            });
+
+            var width = form.$('th[data-name="date"]')[0].style.width;
+
+            await testUtils.dom.click(form.$('.o_field_x2many_list_row_add a'));
+
+            assert.strictEqual(form.$('th[data-name="date"]')[0].style.width, width);
```
Comment (ged-odoo) [inline addons/web/static/tests/fields/relational_fields/field_one2many_tests.js]: should assert that there is at least one row

File: addons/web/static/tests/views/list_tests.js [hunk 3]
```diff
@@ -1517,6 +1517,93 @@ QUnit.module('Views', {
...
+                        '<field name="datetime"/>' +
+                        '<field name="text"/>' +
+                    '</tree>',
+        });
+
+        var width = list.$('th[data-name="datetime"]')[0].style.width;
+
+        await testUtils.dom.click(list.$buttons.find('.o_list_button_add'));
+
+        assert.strictEqual(list.$('th[data-name="datetime"]')[0].style.width, width);
```
Comment (ged-odoo) [inline addons/web/static/tests/views/list_tests.js]: same

File: addons/web/static/tests/views/list_tests.js [hunk 4]
```diff
@@ -1497,6 +1497,26 @@ QUnit.module('Views', {
...
+            View: ListView,
+            model: 'foo',
+            data: this.data,
+            arch: '<tree editable="top">' +
+                        '<field name="datetime" widget="date"/>' +
+                        '<field name="text"/>' +
+                    '</tree>',
+        });
+
+        assert.strictEqual(list.$('th[data-name="datetime"]')[0].style.width, '92px',
```
Comment (ged-odoo) [inline addons/web/static/tests/views/list_tests.js]: what about 93px ? 93 is divisible by 3...

File: doc/reference/views.rst [hunk 5]
```diff
@@ -329,11 +329,13 @@ Possible children elements of the list view are:
...
         field, so e.g. ``invisible`` will hide the field but leave the same
         field of other records visible, it will not hide the column itself
-    ``width_factor`` (for ``editable``)
-        the column relative width (as the layout is fixed)
     ``width`` (for ``editable``)
-        the column width (as the layout is fixed). It is a string describing the
-        width css property, such as '100px'.
+        when there is no data in the list, the width of a column can be forced
+        by setting this attribute. The value can be an absolute width (e.g.
+        '100px'), or a relative weigth (e.g. '3', meaning that this column will
```
Comment (ged-odoo) [inline doc/reference/views.rst]: weight

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by Bonainifo** [comment]
Is it support fixed width of field? such width="60px;"

**Comment by aab-odoo** [comment]
@Bonainifo Yes it does

**Comment by Bonainifo** [comment]
Wow! wonderful

**Pull request closed without merging**